WorkManagerSample/benchmark/src/androidTest/java/com/example/benchmark/WorkerBenchmark.kt

```diff
@@ -0,0 +1,123 @@
+import androidx.test.runner.AndroidJUnit4
+import androidx.work.*
+import androidx.work.impl.utils.SynchronousExecutor
+import androidx.work.testing.WorkManagerTestInitHelper
+import com.example.lib.Constants
+import com.example.lib.workers.BaseFilterWorker
+import com.example.lib.workers.BaseFilterWorker.Companion.inputStreamFor
+import com.example.lib.workers.BlurEffectFilterWorker
+import com.example.lib.workers.GrayScaleFilterWorker
+import com.example.lib.workers.WaterColorFilterWorker
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@SmallTest
+class WorkerBenchmark {
+
+    companion object {
+        private const val JETPACK = "${BaseFilterWorker.ASSET_PREFIX}images/jetpack.png"
+    }
+
+    private lateinit var mContext: Context
+    private lateinit var mTargetContext: Context
+
+    @get:Rule
+    val benchmarkRule = BenchmarkRule()
+
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @Before
+    fun setUp() {
+        mContext = InstrumentationRegistry.getContext()
+        mTargetContext = InstrumentationRegistry.getTargetContext()
+    }
+
+    @Test
+    fun testBlurEffectFilterWorker() {
+        val worker = createWorker { appContext: Context, _: String,
+                                    workerParameters: WorkerParameters ->
+            BlurEffectFilterWorker(appContext, workerParameters)
+        }
+
+        benchmarkRule.measure {
+            val inputStream = inputStreamFor(mContext, JETPACK)
+            worker.applyFilter(BitmapFactory.decodeStream(inputStream))
+            inputStream?.close()
+        }
+    }
+
+    @Test
+    fun testGrayScaleFilterWorker() {
+        val worker = createWorker { appContext: Context, _: String,
+                                    workerParameters: WorkerParameters ->
+            GrayScaleFilterWorker(appContext, workerParameters)
+        }
+
+        benchmarkRule.measure {
+            val inputStream = inputStreamFor(mContext, JETPACK)
+            worker.applyFilter(BitmapFactory.decodeStream(inputStream))
+            inputStream?.close()
+        }
+    }
+
+    @Test
+    fun testWaterColorEffectFilterWorker() {
+        val worker = createWorker { appContext: Context, _: String,
+                                    workerParameters: WorkerParameters ->
+            WaterColorFilterWorker(appContext, workerParameters)
+        }
+
+        benchmarkRule.measure {
+            val inputStream = inputStreamFor(mContext, JETPACK)
+            worker.applyFilter(BitmapFactory.decodeStream(inputStream))
+            inputStream?.close()
+        }
+    }
+
+    private fun <T : ListenableWorker> createWorker(
+            createWorkerBlock: (appContext: Context, workerClassName: String,
+                                workerParameters: WorkerParameters) -> T
+    ): T {
+        lateinit var worker: T
```

Comment:
add a comment about the only way to instantiate a worker is to enqueue work into workmanager

Reply:
:+1: